### PR TITLE
CLI: set build-hook command

### PR DESCRIPTION
**Pull request by csrwng**
Allows setting a post-commit build hook on a build config.

If re-using built image entrypoint:
```
oc set build-hook [bc/]mybc --post-commit -- arg1 arg2
```

If specifying a new entrypoint:
```
oc set build-hook [bc/]mybc --post-commit --commnad -- command arg1 arg2
```

If specifying a script:
```
oc set build-hook [bc/]mybc --post-commit --script="command1 | command2 > output.txt && command3"
```

Although there's only one type of hook is currently allowed, the argument with the type of hook must be specified (`--post-commit`).

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees ptal

**Comment by csrwng** [comment]
[test]

**Comment by csrwng** [comment]
[test]

File: pkg/cmd/cli/cmd/set/buildhook.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+		},
+	}
+
+	kcmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", options.Selector, "Selector (label query) to filter build configs")
+	cmd.Flags().BoolVar(&options.All, "all", options.All, "Select all build configs in the namespace")
+	cmd.Flags().StringSliceVarP(&options.Filenames, "filename", "f", options.Filenames, "Filename, directory, or URL to file to use to edit the resource.")
+
+	cmd.Flags().BoolVar(&options.PostCommit, "post-commit", options.PostCommit, "If true, set the post-commit build hook on a build config")
+	cmd.Flags().BoolVar(&options.Entrypoint, "command", options.Entrypoint, "If true, set the entrypoint of the hook container to the given command")
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/set/buildhook.go]: I would prefer to use the same syntax as for exec probes here --

Comment (smarterclayton) [inline pkg/cmd/cli/cmd/set/buildhook.go]: Hrm, wondering if people expect this to be the default.

Comment (smarterclayton) [inline pkg/cmd/cli/cmd/set/buildhook.go]: I guess this is fine.

**Comment by bparees** [comment]
lgtm [merge]

**Pull request merged**